src/generate_perturbation_derivatives.jl

```diff
@@ -366,14 +361,15 @@ function ChainRulesCore.rrule(::typeof(generate_perturbation), m::PerturbationMo
                     Δp[i] += dot(c.B_p[i], Δsol.B)
                 end
             end
-            if (~isnothing(Δsol.D)) # D is a Distribution object which complicates stuff here
-                if ((Δsol.D != NoTangent()) & (Δsol.D != ZeroTangent()))
-                    # Only supports diagonal matrices for now.
-                    for i in 1:n_p_d
-                        Δp[i] += dot(c.Ω_p[i], Δsol.D.σ)
-                    end
-                end
-            end
+            # Since we are not going to differentiate on D ever, we'll skip it
```

Comment:
I just want to make sure that if we are removing this we need to have error handling.  Should it fail when it sees `(Δsol.D != ZeroTangent())` or something like that.

Reply:
No, it's because the constructor changed: it treats the vector as standard deviations before, but now as variances. Hence the algebra is also subject to change. Although the algebra is simple, I would rather just mute it at this point in case there are future changes in the structure of `D`.

Reply:
It isn't possible to throw an error to prevent misuse?

Reply:
Otherwise this looks like a big future bug waiting to happen in ours or someone else's code

Reply:
I'll uncomment it and deal with the algebra then.

Reply:
Or just put in a tiny error check?